lib/index_documents.rb

```diff
@@ -0,0 +1,65 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    links = message.payload["links"].try(:dup)
+
+    if links
+      links.each do |(k, vs)|
+        paths = vs.map { |v| base_path(v) }.compact
+        links[k] = paths
+      end
+
+      base_path = document_base_path(message)
+      update_index(base_path, links)
+    end
+
+    message.ack
+  rescue StandardError => e
```

Comment:
What would happen if we didn't do this rescue at all?

I think that errors will then be caught by the message queue consumer gem that called this method, the consumer will exit (and report the error) and then upstart will restart it and the message will be reprocessed.  This would be functionally very similar to this code, but would also trigger alerting if a message was failing too frequently (because the process would restart frequently, too).  It would also mean less code to test and maintain.